### PR TITLE
Next

### PR DESCRIPTION
**Pull request by RanjaniOS**
1.Protobuf issues fixed.
2.Auto connect and reconnect.
3.code refactor for BLE,Tracefile,NetworkData and RxData parser.
4.Timestamp in diagnostic message.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -1,4 +0,0 @@
-[submodule "protobuf-swift"]
```
Comment (kani19) [inline .gitmodules:1]: There is nothing in gitignore.

File: CHANGELOG.mkd [hunk 2]
```diff
@@ -1,4 +1,21 @@
 # OpenXC iOS Library Changelog
+## v2.1.0
+* Feature: Data source selection in Settings Page.
+* Feature: Custom command added in Commands Page
+* Feature: Diagnostic and command message support added in Protobuf mode.
+* Feature: Manual format change support added. JSON to proto & vice-versa can be changed in commands page
+* Feature: Disconnect C5BLE from application
```
Comment (kani19) [inline CHANGELOG.mkd]: Add feature - autoconnect & reconnect

File: openXCiOSFrameworkTests/NetworkDataTest.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
+//
+//  NetworkDataTest.swift
```
Comment (kani19) [inline openXCiOSFrameworkTests/NetworkDataTest.swift:2]: As discussed, please delete the test classes for now. and once you are done with Unit Testing, you can include the correct classes

**Review by kani19 — COMMENTED** [review]
Requires a lot of testing before we merge

**Review by emarsman — APPROVED** [review]
Ready for release

**Pull request merged**